### PR TITLE
[NCL-4996] implemented tests for BCProvider

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [x] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    public void testStore() {
+        final String name = "NewHire";
+        final String repoId = "12";
+        final String envId = "12";
+        final String projId = "12";
+
+        org.jboss.pnc.dto.BuildConfiguration buildConfiguration = org.jboss.pnc.dto.BuildConfiguration.builder()
+                .name(name)
+                .buildType(BuildType.MVN)
+                .project(org.jboss.pnc.dto.Project.builder().id(projId).build())
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java]: Wouldn't ProjectRef be technically better?

File: facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        org.jboss.pnc.dto.BuildConfiguration refreshed = provider.getSpecific(dependency.getId());
+        assertThat(refreshed.getDependencies())
+                .anySatisfy(config -> assertThat(config.getId()).isEqualTo(bc.getId().toString()));
+    }
+
+    @Test
+    public void testRemoveDependency() {
+        //With
+        org.jboss.pnc.dto.BuildConfiguration dependency = provider.getSpecific("2"); //BC(name: "First!")
+        org.jboss.pnc.dto.BuildConfiguration dependant = provider.getSpecific("3"); //BC(name: "Second!!")
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java]: not confusing at all ;-)

File: facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java (hunk 1 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java]: Huh, ya.. you are right. Full project dto can be misleading.

File: facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java (hunk 2 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/BuildConfigProviderTest.java]: any ideas for improvement? Change naming to child and parent?

**Pull request merged**